### ISSUE TITLE
Cooling temp wont stay at 75F.....

### ISSUE THREAD

**Issue by SkinnyT81** [georgezhao2010/midea_ac_lan#94]
When changing the cooling temp on my Midea unit and setting it to 75F, the climate card seems to pause and then it changes itself to 73F.    I can manually set it to 74F and 76F.   I can post the logs sometime tomorrow.

**Comment by georgezhao2010** [comment]
> When changing the cooling temp on my Midea unit and setting it to 75F, the climate card seems to pause and then it changes itself to 73F. I can manually set it to 74F and 76F. I can post the logs sometime tomorrow.

Midea AC setting the temperature in Celsius and convert to Fahrenheit by HA. The two can not be one-to-one correspondence, so some temperature values in Fahrenheit cannot be set.

**Comment by georgezhao2010** [comment]
I see what the problem is.

When you set the temperature to 75F, HA convert it to 23.89C and transfer to me.  I send 23.89C to AC， AC set the temperature to 23C (73.4F), HA showing it as 73F.

Strangely, when you set  temperature to 74F(23.33C), AC will set the temperature to 23.5C(74.3F) and HA will showing it as 74F, it looks correct.

I found a way to fix it although it's not my fault.

**Comment by SkinnyT81** [comment]
Awesome!   I wasn't sure if it had something to do with the fact that HA reports everything in Fahrenheit but the Midea AC display shows in Celsius.   I can change it from the app, but anytime I change the temp from HA, the display goes back to Celsius.   On one of the earlier versions, I changed something in either device.py or climate.py and the setting would stick until an update.

**Comment by SkinnyT81** [comment]
2022-09-15 06:50:30.368 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform midea_ac_lan.fan: Platform not found (No module named 'custom_components.midea_ac_lan.midea.devices.fa').

I'm now seeing this in my logs after the last update.

**Comment by chunkystyles** [comment]
For anyone seeing this issue, I solved this in a hacky way.  I created buttons for adjusting temperature in Node-RED.  From those buttons, when I go to do the temperature adjustment, I run it through a mapping of values that I know will give me the desired temperature.  I figured out the mapping through just trial and error.

{
    80: 80,
    79: 79,
    78: 78,
    77: 77.2,
    76: 76,
    75: 75.3,
    74: 74,
    73: 73.5,
    72: 72,
    71: 71,
    70: 70,
    69: 69,
    68: 68,
    67: 67,
    66: 66.5,
    65: 65,
    64: 63.5,
    63: 63,
    62: 62,
    61: 60.8
}
